legate/core/operation.py

```diff
@@ -151,27 +170,35 @@ def add_alignment(self, store1, store2):
         part2 = self._get_unique_partition(store2)
         self.add_constraint(part1 == part2)
 
-    def add_broadcast(self, store, axes=None):
+    def add_broadcast(
+        self, store: Store, axes: Optional[Union[int, Iterable[int]]] = None
+    ) -> None:
         self._check_store(store)
-        part = self._get_unique_partition(store)
+        part = self._get_unique_partition(store)  # XXXXX
         self.add_constraint(part.broadcast(axes=axes))
 
-    def add_constraint(self, constraint):
+    def add_constraint(self, constraint: Constraint) -> None:
         self._constraints.append(constraint)
 
-    def execute(self):
+    def execute(self) -> None:
         self._context.runtime.submit(self)
 
-    def get_tag(self, strategy, part):
+    def get_tag(self, strategy: Strategy, part: Any) -> Literal[1, 0]:
```

Comment:
Your choice of the return type is rather interesting. (Does MyPy support dependent types?) What's the reason to favor this over `int`?

Reply:
I can handle [a few kinds of `Literal`](https://mypy.readthedocs.io/en/stable/literal_types.html#literal-types), at least. 

Something (I forget) suggested to me that it was only 0 and 1 so I was trying to be as narrow as possible. Can there be other values besides 0 and 1? If so then it for sure should be change. If not, then it's probably fine either way, per preference.

Reply:
It also seems an enum could potentially be another option here

Reply:
changed https://github.com/nv-legate/legate.core/pull/189/commits/99257dbf2be5778ac6fe2febfa88cf35ab1e2673